js/3d/top-view.js

```diff
@@ -62,6 +69,10 @@ export default class {
     if (earthquake) {
       return { type: 'earthquake', data: earthquake }
     }
+    const eruption = this.eruptions.spriteAt(x, this._height - y)
+    if (eruption) {
+      return { type: 'eruption', data: eruption }
+    }
```

Comment:
I think it'd be better if that's the first statement in this function. So, an eruption click is detected before volcano.